src/main/java/edu/indiana/dlib/amppd/controller/BundleItemController.java

```diff
@@ -28,23 +31,41 @@
     private ItemRepository itemRepository;
 
 	@PostMapping("/bundles/{bundleId}/add/items/{itemId}")
-    public Bundle addItemToBundle(@PathVariable("bundleId") Long bundleId, @PathVariable("itemId") Long itemId) {		
+	public Bundle addItemToBundle(@PathVariable("bundleId") Long bundleId, @PathVariable("itemId") Long itemId) {		
 		Item item = itemRepository.findById(itemId).orElseThrow(() -> new StorageException("item <" + itemId + "> does not exist!"));    
 		Bundle bundle = bundleRepository.findById(bundleId).orElseThrow(() -> new StorageException("bundle <" + bundleId + "> does not exist!"));    
+
+		// don't add the item if it already exists in the bundle
+		if (bundle.getItems().contains(item)) {
+			// TODO the warning message shall be displayed back to the screen
+			String msg = "Item <" + itemId + "> has already been added to bundle<" + bundleId + ">!";
+			log.warning(msg);	
+			return bundle;
+		}
+
 		item.getBundles().add(bundle);	// need to add from item side since Item owns the M;M relationship
 		bundle.getItems().add(item);	// TODO do we need this?
```

Comment:
I feel we don't need this. Adding bundle to item is sufficient to establish a M:M relationship.